src/TraceEvent/Computers/StartStopThreadTimeComputer.cs

```diff
@@ -0,0 +1,229 @@
+        }
+        
+        protected override void RecordAdditionalDataOnMissingStopAtTraceEnd(StackSourceSample sample)
+        {
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {sample.Metric*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+        
+        protected override void RecordAdditionalStopData(StackSourceSample sample, TraceEvent data)
+        {
+            var stopData = (ContentionStopTraceData) data;
+            sample.Metric = (float) (stopData.DurationNs / NanosInMillisecond);
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {stopData.DurationNs:N0}"), sample.StackIndex);
+        }
+    }
+    
+    public class WaitHandleWaitTimeComputer : StartStopThreadTimeComputer
+    {
+        public WaitHandleWaitTimeComputer(TraceLog eventLog, MutableTraceEventStackSource stackSource) : base(eventLog, stackSource)
+        {}
+
+        protected override void Subscribe(TraceLogEventSource eventSource)
+        {
+            eventSource.Clr.WaitHandleWaitStart += OnStart;
+            eventSource.Clr.WaitHandleWaitStop += OnStop;
+        }
+
+        protected override void RecordAdditionalDataOnMissingStartAtTraceStart(StackSourceSample sample)
+        {
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {sample.Metric*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalStartData(StackSourceSample sample, TraceEvent data)
+        {
+            var startData = (WaitHandleWaitStartTraceData) data;
+            sample.StackIndex = _interner.CallStackIntern(_interner.FrameIntern($"EventData WaitSource {startData.WaitSource}"), sample.StackIndex);
+            sample.StackIndex = _interner.CallStackIntern(_interner.FrameIntern($"EventData AssociatedObjectID {startData.AssociatedObjectID}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalDataOnMissingStopAtTraceEnd(StackSourceSample sample)
+        {
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {sample.Metric*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+
+        protected override void RecordAdditionalStopData(StackSourceSample sample, TraceEvent data)
+        {
+            var stopData = (WaitHandleWaitStopTraceData) data;
+            var durationMs = stopData.TimeStampRelativeMSec - sample.TimeRelativeMSec; // recompute the duration to get better precision than float
+            sample.StackIndex = _stackSource.Interner.CallStackIntern(_stackSource.Interner.FrameIntern($"EventData DurationNs {durationMs*NanosInMillisecond:N0}"), sample.StackIndex);
+        }
+    }
+    
+    public abstract class StartStopThreadTimeComputer
```

Comment:
I like the idea of what you're doing here.  I would recommend calling this something different, since it's not actually a thread time computer, and we have a thread time computer already.  I think I would call this something like `StartStopLatencyComputer`, since it calculates the latency between starts and stops.

Reply:
Done